packages/ai/src/ai/web/endpoints/webhook.py

```
@@ -0,0 +1,170 @@
+        payload: Raw request body bytes.
+        signature: The signature provided in the X-Webhook-Signature header.
+        secret: The shared webhook secret.
+
+    Returns:
+        True if the signature is valid, False otherwise.
+    """
+    expected = hmac.new(secret.encode('utf-8'), payload, hashlib.sha256).hexdigest()
+    return hmac.compare_digest(expected, signature)
+
+
+async def webhook_trigger(request: Request, pipeline_id: str, body: Optional[Dict[str, Any]] = None) -> Result:
```

Ruff (0.15.7)

[warning] Too many return statements (7 > 6)

(PLR0911)